run-build-functions.sh

```diff
@@ -614,13 +631,30 @@ install_dependencies() {
 
     restore_home_cache ".node/corepack" "corepack dependencies"
 
-    if [ "$NETLIFY_USE_YARN" = "true" ] || ([ "$NETLIFY_USE_YARN" != "false" ] && [ -f yarn.lock ]); then
-      run_yarn $YARN_VERSION "$featureFlags"
-    elif [ "$NETLIFY_USE_PNPM" = "true" ] || ([ "$NETLIFY_USE_PNPM" != "false" ] && [ -f pnpm-lock.yaml ]); then
-      run_pnpm $PNPM_VERSION "$featureFlags"
-    else
-      run_npm "$featureFlags"
-    fi
+		if has_feature_flag "$featureFlags" "build-image_use_new_package_manager_detection"; then
+			local pkgManager=$(echo "$buildInfo" | jq -r '.packageManager.name')
```

Comment:
We should apply the same strategy here of piping or using `jq`'s stderr logs.

Reply:
Same as already mentioned earlier.